### DISCUSSION TITLE
DAP for R

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [R-nvim/R.nvim#79] [Q&A]
The removal of the debugging facilities let me wonder: is there a DAP implementation for R?

**Comment by PMassicotte** [comment]
Nothing implemented at the moment. Might be a good idea to start thinking about it.

**Comment by jalvesaq** [comment]
I would be glad to see it implemented. Unfortunately, I don't have time to work on this anytime soon. I'm prioritizing fixing bugs in existing features and optimizing code already written.

**Reply by raffaelemancuso** [reply]
from what I saw in debugpy, it looks like a non trivial task

**Comment by januz** [comment]
some related work: https://github.com/dgkf/debugadapter

**Comment by jalvesaq** [comment]
The last commit was five months ago, and the README says: _Currently, nothing will happen. Communication about the current debugger state back to the client is in early development._
